src/ns/core/notifier.py

```diff
@@ -58,20 +59,31 @@ def __init__(
         *,
         config: NotifierConfig,
         smtp_client: SmtpClientPort,
+        sms_client: SmsClientPort,
     ):
         """Initialize the Notifier with configuration and smtp client"""
         self._config = config
         self._smtp_client = smtp_client
+        self._sms_client = sms_client
 
-    async def send_notification(
+    async def send_email_notification(
         self,
         *,
-        notification: event_schemas.Notification,
+        notification: event_schemas.EmailNotification,
     ):
         """Sends out notifications based on the event details"""
         message = self._construct_email(notification=notification)
         self._smtp_client.send_email_message(message)
 
+    async def send_sms_notification(
+        self,
+        *,
+        notification: event_schemas.SmsNotification,
+    ):
+        """Sends out notifications based on the event details"""
+        message = notification.model_dump()
+        self._sms_client.send_sms_message(phone=message["phone"], text=message["text"])
```

Comment:
```suggestion
        self._sms_client.send_sms_message(phone=notification.phone, text=notification.text)
```
The pydantic model doesn't need to be dumped, you can access it directly.